src/test/java/com/google/cloud/tools/maven/deploy/DeployerTest.java

```diff
@@ -0,0 +1,103 @@
+package com.google.cloud.tools.maven.deploy;
+
+import static org.mockito.Mockito.times;
+
+import com.google.cloud.tools.appengine.configuration.DeployConfiguration;
+import com.google.cloud.tools.appengine.configuration.DeployProjectConfigurationConfiguration;
+import com.google.cloud.tools.appengine.operations.Deployment;
+import com.google.cloud.tools.maven.cloudsdk.CloudSdkAppEngineFactory;
+import com.google.cloud.tools.maven.deploy.AppDeployer.ConfigBuilder;
+import com.google.cloud.tools.maven.stage.AppEngineWebXmlStager;
+import com.google.cloud.tools.maven.stage.AppYamlStager;
+import com.google.cloud.tools.maven.stage.Stager;
+import java.io.IOException;
+import java.nio.file.Path;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DeployerTest {
+
+  @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
```

Comment:
`final`

Reply:
Generally haven't been doing this, but I can do it.